crates/metadata/src/layout/validate.rs

```diff
@@ -0,0 +1,298 @@
+// Copyright 2018-2022 Parity Technologies (UK) Ltd.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::layout::{
+    Layout,
+    MetadataError,
+    StructLayout,
+};
+use ink_prelude::collections::HashMap;
+use ink_primitives::Key;
+use scale_info::form::MetaForm;
+
+/// It validates that the storage layout:
+/// - Hasn't conflicting storage keys, otherwise returns an error with a description of the conflict
```

Comment:
```suggestion
/// It validates that the storage layout doesn't have conflicting storage keys.
/// Otherwise an error with a description of the conflict is returned.
```